src/huggingface/forte/huggingface/question_and_answering_multi.py

```diff
@@ -0,0 +1,147 @@
+        )
+        for i, result in enumerate(result_collection):
+            start = result["start"]
+            end = result["end"]
+            doc_pack = input_pack.get_pack(doc_id_list[i])
+            ans_phrase = Phrase(pack=doc_pack, begin=start, end=end)
+            input_pack.add_entry(
+                MultiPackLink(input_pack, first_question, ans_phrase)
+            )
+
+    @classmethod
+    def default_configs(cls):
+        r"""This defines a basic config structure for `QuestionAnsweringSingle`.
+
+        Following are the keys for this dictionary:
+            - `entry_type`: defines which entry type in the input pack to make
+              prediction on. The default makes prediction on each `Document`
+              in the input pack.
+            - `model_name`: language model, default is
+              `"ktrapeznikov/biobert_v1.1_pubmed_squad_v2"`.
+              The wrapper supports Hugging Face models with pipeline tag of
+              `question-answering`.
+            - `max_answer_len`: The maximum length of predicted answers (e.g.,
+              only answers with a shorter length are considered).
+            - `cuda_device`: Device ordinal for CPU/GPU supports. Setting
+              this to -1 will leverage CPU, a positive will run the model
+              on the associated CUDA device id.
+            - `handle_impossible_answer`: Whether or not we accept
+              impossible as an answer.
+
+        Returns: A dictionary with the default config for this processor.
+        """
+        config = super().default_configs()
+        config.update(
+            {
+                "question_pack_name": "question",
+                "entry_type": "ft.onto.base_ontology.Document",
+                "model_name": "ktrapeznikov/biobert_v1.1_pubmed_squad_v2",
+                "max_answer_len": 15,
+                "cuda_devices": -1,
+                "handle_impossible_answer": False,
+            }
+        )
+        return config
+
+    @classmethod
+    def expected_types_and_attributes(cls):
+        r"""Method to add user specified expected type which
+        would be checked before running the processor if
+        the pipeline is initialized with
+        `enforce_consistency=True` or
+        :meth:`~forte.pipeline.Pipeline.enforce_consistency` was enabled for
+        the pipeline.
+        """
+        return {cls().default_configs()["entry_type"]: set()}
+
+    def record(self, record_meta: Dict[str, Set[str]]):
+        r"""Method to add output type record of `QuestionAnsweringSingle` which
+        is `"ft.onto.base_ontology.Phrase"`
+        to :attr:`forte.data.data_pack.Meta.record`.
+
+        Args:
+            record_meta: the field in the datapack for type record that need to
+                fill in for consistency checking.
+        """
+        if "ft.onto.base_ontology.Phrase" not in record_meta.keys():
+            record_meta["ft.onto.base_ontology.Phrase"] = set()
```

Comment:
Looks like this actually does type check on the `MultiPack` level, which means, we need to have `MultiPackLink` for the multipack, and `Phrase` for the single pack. Any ideas on this?